src/main/java/jenkins/plugins/slack/StandardSlackService.java

```diff
@@ -88,6 +101,26 @@ public boolean publish(String message, String color) {
         return result;
     }
 
+    private String getTokenToUse() {
+        if (authTokenCredentialId != null && !authTokenCredentialId.isEmpty()) {
+            StringCredentials credentials = lookupCredentials(authTokenCredentialId);
+            if (credentials != null) {
+                logger.info("Using Integration Token Credential ID.");
+                return credentials.getSecret().getPlainText();
+            }
+        }
+
+        logger.info("Using Integration Token.");
+
+        return token;
+    }
+
+    private StringCredentials lookupCredentials(String credentialId) {
+        List<StringCredentials> credentials = CredentialsProvider.lookupCredentials(StringCredentials.class, Jenkins.getInstance(), ACL.SYSTEM, Collections.<DomainRequirement>emptyList());
```

Comment:
The particular snippet you posted won't work because there isn't a `build` parameter in the `StandardSlackService` class and I also think it would be outside the scope of the class to care about the build.  Are you mainly just interested in throwing the `CredentialNotFoundException`?  I'm sure I can work that in if it can't find the credentials.

If you were to give it the ID of a credential other than type `StringCredentials` it would return `null` and continue on its merry way.  Maybe I'm not following the functionality you're looking for with the snippet you posted.